jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/core/WebSocketCloseTest.java

```diff
@@ -190,14 +186,27 @@ public void serverFailClose_ISHUT() throws Exception
         setup(State.ISHUT);
         server.handler.receivedCallback.poll().failed(new Exception("test failure"));
 
-        Frame frame = receiveFrame(client.getInputStream());
-        assertNotNull(frame);
-        assertThat(new CloseStatus(frame.getPayload()).getCode(), is(CloseStatus.SERVER_ERROR));
+        CloseStatus closeStatus = new CloseStatus(receiveFrame(client.getInputStream()));
+        assertThat(closeStatus.getCode(), is(CloseStatus.SERVER_ERROR));
+        assertThat(closeStatus.getReason(), is("test failure"));
 
         assertTrue(server.handler.closed.await(10, TimeUnit.SECONDS));
         assertThat(server.handler.closeStatus.getCode(), is(CloseStatus.SERVER_ERROR));
     }
 
+    @Test
+    public void clientClosesOutput_ISHUT() throws Exception
```

Comment:
We need to parameterise this test harness to also test all these cases over WSS (ie SSL)